### PR TITLE
Dialect agnostic CSE Pass

### PR DESCRIPTION
**Pull request by yushangdi**
Fixes comments in https://github.com/pytorch/pytorch/pull/81512

- banned ops is an input to the pass
- update the fx/readme.md to include this file for better discoverability
- use make_fx in torch repo

### CODE REVIEW

**Review by SherlockNoMad — APPROVED** [review]
per comment.

File: torch/fx/passes/dialect/common/cse_pass.py [hunk 1]
```diff
@@ -13,22 +13,34 @@ def get_aten_target(node):
...
+
+def get_CSE_banned_ops():
+    return rand_ops.union(inplace_ops)
 
 
 class CSEPass(PassBase):
 
+    def __init__(self, banned_ops=None):
+        """
+        Constructor for CSEPass. Users can specify a list of ops to ban from CSE.
```
[the remark this answers is not in the document]
Comment (yushangdi) [reply torch/fx/passes/dialect/common/cse_pass.py]: fixed

**Pull request closed without merging**